### ISSUE TITLE
Non-ascii characters are not shown

### ISSUE THREAD

**Issue by piranna** [outsinre/cjktty-patch#3]
I've managed to apply the patch and compile the font statically on the kernel and use it by default, but when using a non-ascii character (more particularly, braille patterns) they are not shown and instead is a space, as if the characters where transparent. Do you know if I need to enable something or define an environment variable?

**Comment by outsinre** [comment]
@piranna afaik, cjktty patch does not handle chracters beyond CJK. You may check your system fonts setting, whether symbols fonts are installed.

I am not pretty sure. If you get it, welcome to note me :)

**Comment by piranna** [comment]
Braille patterns are included on Unicode BMP and I have check that the patch include them by inspecting manually that they are there, so there should be a problem in other place. I have tried by setting LANG environment variable to one using UTF-8  without luck, and the same, and the same for enabling unicode support in the terminal using the special ansi code used by unicode_start script, so I don't know if I need to do something more... :-/

**Comment by piranna** [comment]
In fact, in all the cases I have tried it, sending escaped values to the terminal to print CJK glyphs, I get instead characters of the extended ascii set :-/ I know the cjktty font is loaded because it's the only one included in the kernel, so this is not the problem. On a side note, I've found the terminal drawing is fairly slower when using the cjktty font (there's a minor delay when typing, and moving the cursor on the cli readline with the arrow keys is anoying slow compared with normal fonts), do you know why it would happeng and if it could be fixed?

**Comment by outsinre** [comment]
Unicode BMP has many planes. Are you sure the specific braille pattern BMP included?

I have no such cjktty patch delay.

You could give me an example to check whether braille patterns are shown on my console.

**Comment by piranna** [comment]
> Unicode BMP has many planes. Are you sure the specific braille pattern
> BMP included?

https://en.m.wikipedia.org/wiki/Braille_Patterns

This you have more info about them. I have also check for the braille
pattern range on the patch and was able to find them on their location...

> I have no such cjktty patch delay.

I have given it more attention and seems it has the same speed, but on the
estandar terminal the cursor seems to "slide" while with the cjktty font is
mostly like jumping between characters. Maybe it's an issue of QEmu that
didn't notice before, or just a mental trick of ny own...

> You could give me an example to check whether braille patterns are shown
> on my console?

Drawille package use braille patterns to draw lines on the terminal. A
simple way would be to try to print any of their escaped unicode keypoints
on the terminal with printf or similar.

**Comment by outsinre** [comment]
So it probably has nothing to do with cjktty patch.

**Comment by piranna** [comment]
> So it probably has nothing to do with cjktty patch.

Yes, I'm not saying that it's a problem of the patch, but instead about how could I be able to enable it and being able to show unicode characters on the console... So far I've just only be able to compile the kernel with the patch and use the font (since there's no other available), but don't know if I need to do any other step... :-/ Setting the `$LANG` environment variable to `es_ES.UTF-8` (Spain unicode locale) doesn't work, and enabling the Linux console Unicode mode with `echo -n -e '\033%G'` (https://kernel.googlesource.com/pub/scm/linux/kernel/git/legion/kbd/+/1.10/src/unicode_start) doesn't work (and in any case it's enabled by default on the Linux kernel source code...). Am I missing any other step?

**Comment by outsinre** [comment]
Have you tried to set LANG to any one of CJK?

**Comment by piranna** [comment]
> Have you tried to set LANG to any one of CJK?

No... can you give one that I can be able to use, for example yours?

**Comment by outsinre** [comment]
Try this one `ja_JP.UTF-8 UTF-8`

**Comment by piranna** [comment]
> Try this one ja_JP.UTF-8 UTF-8

No result :-/ Don't know if it would be a problem of Node.js...
![pantallazo-2016-09-29 16-58-00](https://cloud.githubusercontent.com/assets/532414/18959300/f610f958-8665-11e6-8051-804d27216ec2.png)

**Comment by outsinre** [comment]
maybe. I think it's not the patch issue.
